Arrangement_on_surface_2/include/CGAL/Arr_polyline_traits_2.h

```diff
@@ -437,11 +437,17 @@ class Arr_polyline_traits_2 : public Arr_polycurve_traits_2<SegmentTraits_2> {
                               "Cannot construct a degenerated segment");
         return Subcurve_2(boost::get<0>(t), boost::get<1>(t));
       };
+      auto begin_plus_one(begin);
+      ++begin_plus_one;
+      auto end_minus_one(end);
+      --end_minus_one;
       return operator()(boost::make_transform_iterator
-                        (make_zip_iterator(boost::make_tuple(begin, begin + 1)),
+                        (make_zip_iterator(boost::make_tuple(begin,
+                                                             begin_plus_one)),
                          point_pair_to_segment),
                         boost::make_transform_iterator
-                        (make_zip_iterator(boost::make_tuple(end - 1, end)),
+                        (make_zip_iterator(boost::make_tuple(end_minus_one,
+                                                             end)),
                          point_pair_to_segment));
```

Comment:
Hi @efifogel,

Next time you need that, think about [`std::next`](https://en.cppreference.com/w/cpp/iterator/next) and [`std::prev`](https://en.cppreference.com/w/cpp/iterator/prev), that are really useful for that sort of thing.